### PR TITLE
Analyzer Tests

### PR DESCRIPTION
**Pull request by Vijay-P**
**What this PR does / why we need it**:
Adds functional tests to for the remaining analyzer modules that will be replaced by syft. These tests will allow us to catch regressions after the syft integration is complete.

**Which issue this PR fixes**: 

- ENTERPRISE-81
- ENTERPRISE-83
- ENTERPRISE-84
- ENTERPRISE-91
- ENTERPRISE-92

**Special notes**:
Includes tests for:
* anchore_engine/analyzers/modules/13_retrieve_files.py
* anchore_engine/analyzers/modules/13_secret_content_search.py
* anchore_engine/analyzers/modules/32_golang_packages.py
* anchore_engine/analyzers/modules/40_file_suids.py

Also includes a fix for a bug that prevented golang hints from getting processed.

### CODE REVIEW

File: anchore_engine/analyzers/manager.py [hunk 1]
```diff
@@ -29,8 +29,8 @@ def apply_hints(analyzer_report, unpackdir):
     # the config does not explicitly enable hints processing, effectively disabling
     # hints processing.
     for engine_entry in utils.content_hints(unpackdir=unpackdir):
-        pkg_type = engine_entry.get("type")
-        if pkg_type:
+        pkg_type = engine_entry.get("type", "")
+        if pkg_type and pkg_type != "go":
             handler = syft.modules_by_engine_type[pkg_type]
```
Comment (Vijay-P) [inline anchore_engine/analyzers/manager.py:34]: This was failing with KeyError when pkg_type == "go", causing all go hints to fail.

File: tests/functional/clients/conftest.py [hunk 2]
```diff
@@ -97,6 +97,22 @@
...
     },
+    "suids": {
+        "registry": "docker.io",
+        "repo": "anchore/test_images",
+        "tag": "suids",
+        "digest": "sha256:1d0df8e380b947e9f76a1082cc550c3634dbbcfeb78e4c4874eeb149f377326d",
+        "image_source": "registry",
+        "schema_version": "v2",
+    },
+    "secrets": {
```
Comment (Vijay-P) [inline tests/functional/clients/conftest.py:108]: new image with secrets files

File: tests/functional/clients/conftest.py [hunk 3]
```diff
@@ -97,6 +97,22 @@
         "image_source": "registry",
         "schema_version": "v2",
     },
+    "suids": {
```
Comment (Vijay-P) [inline tests/functional/clients/conftest.py:100]: new image with chmod u+s on a file

File: tests/functional/clients/scripts/standalone.py [hunk 4]
```diff
@@ -162,10 +163,15 @@ def create_directories(work_dir):
...
+    sub_directories = [service_dir, "cache_dir"]
     for _dir in sub_directories:
         os.makedirs(join(work_dir, _dir), exist_ok=True)
-
+    # add analyzer config file
+    current_path = os.path.dirname(os.path.realpath(__file__))
+    config_source = join(current_path, "analyzer_config.yaml")
+    config_dest = join(work_dir, service_dir, "analyzer_config.yaml")
+    shutil.copyfile(config_source, config_dest)
     # if work_dir changed, return it so that it can be re-used
```
Comment (Vijay-P) [inline tests/functional/clients/scripts/standalone.py:175]: secret analyzer won't work without a analyzer_config.yaml file

File: tests/functional/clients/standalone/test_hints.py [hunk 5]
```diff
@@ -224,3 +227,25 @@ def test_gem_hints(self, hints_image):
...
+            "packages": [
+                {
+                    "name": "kind",
+                    "version": "v0.10.0",
+                    "type": "go",
+                    "license": "Apache2.0",
+                },
+            ]
+        }
+        result = hints_image(hints, "lean")
```
Comment (Vijay-P) [inline tests/functional/clients/standalone/test_hints.py:244]: can use any image, as the hints file doesn't really do anything. just creates fake entries. so i didn't actually load the go binary in the image.

File: anchore_engine/analyzers/manager.py [hunk 6]
```diff
@@ -29,8 +29,8 @@ def apply_hints(analyzer_report, unpackdir):
     # the config does not explicitly enable hints processing, effectively disabling
     # hints processing.
     for engine_entry in utils.content_hints(unpackdir=unpackdir):
-        pkg_type = engine_entry.get("type")
-        if pkg_type:
+        pkg_type = engine_entry.get("type", "")
```
Comment (nightfurys) [inline anchore_engine/analyzers/manager.py:32]: dictionary .get() returnes `None` if the key doesn't exist. Is there any particular reason why the default value is an empty string?

File: anchore_engine/analyzers/manager.py [hunk 7]
```diff
@@ -29,8 +29,8 @@ def apply_hints(analyzer_report, unpackdir):
...
+        if pkg_type and pkg_type != "go":
```
Comment (nightfurys) [inline anchore_engine/analyzers/manager.py:33]: I see why this was throwing an error for `go` before. But a better check would be to test if pkg_type key exists in syft.modules_by_engine_type. Future syft integrations will add more package types. Limiting on a certain package type (such as go) will mean it has to be changed once a mapper for that package type is available

```suggestion
        if pkg_type and pkg_type in syft.modules_by_engine_type:
```

**Review by nightfurys — CHANGES_REQUESTED** [review]

File: anchore_engine/analyzers/manager.py (hunk 7 above)
Comment (Vijay-P) [reply anchore_engine/analyzers/manager.py]: fixed

File: anchore_engine/analyzers/manager.py (hunk 6 above)
Comment (Vijay-P) [reply anchore_engine/analyzers/manager.py]: fixed

**Review by nightfurys — APPROVED** [review]
lgtm. Rebase with master before merging

**Pull request merged**